DataAccess/Readers/PerfumeReader.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Dapper;
+using DataAccess.DbAccess;
+using DataAccess.Models;
+using Microsoft.Extensions.Configuration;
+
+namespace DataAccess.Readers
+{
+    public class PerfumeReader : IReadPerfume
+    {
+        private readonly PostgreSqlConfiguration _configuration;
+        private readonly IPostgreSqlConnection _buildPostgreSqlConnection;
+        public PerfumeReader(IConfiguration config)
+        {
+            _configuration = new PostgreSqlConfiguration(config);
+            _buildPostgreSqlConnection = new PostgreSqlConnection(_configuration);
+        }
+        public async Task<IEnumerable<Perfume>> GetPerfumes()
+        {
+            await using var connection = _buildPostgreSqlConnection.GetSqlConnection();
```

Comment:
you can implement service for PostgreSql that contains QueryAsync<T> QueryFirstAsync<T> something like this, and use it here in PerfumeReader